code/modules/vehicles/secway.dm

```diff
@@ -28,6 +31,14 @@
 	smoke.start()
 
 /obj/vehicle/ridden/secway/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/reagent_containers/food/snacks/grown/banana))
+		// ignore the occupants because they're presumably too distracted to notice the guy stuffing fruit into their vehicle's exhaust. do segways have exhausts? they do now!
```

Comment:
check if it's nodrop via `!user.transferItemToLoc(W, src)` before you do any of the stuff.